rotkehlchen/tests/exchanges/test_ftx.py

```diff
@@ -0,0 +1,567 @@
+from http import HTTPStatus
+from json.decoder import JSONDecodeError
+from unittest.mock import patch
+import warnings as test_warnings
+
+import pytest
+import requests
+
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.assets.converters import asset_from_ftx
+from rotkehlchen.constants.assets import A_ETH, A_USD, A_USDC
+from rotkehlchen.errors import RemoteError, UnknownAsset, UnsupportedAsset
+from rotkehlchen.exchanges.ftx import FTX
+from rotkehlchen.exchanges.data_structures import AssetMovement, Trade
+from rotkehlchen.fval import FVal
+from rotkehlchen.tests.utils.mock import MockResponse
+from rotkehlchen.typing import AssetMovementCategory, Location, TradeType
+from rotkehlchen.utils.serialization import jsonloads_dict
+
+
+TEST_END_TS = 1617382780
+
+
+def test_name():
+    exchange = FTX('a', b'a', object(), object())
+    assert exchange.name == 'FTX'
+
+
+def test_ftx_exchange_assets_are_known(mock_ftx):
+    request_url = f'{mock_ftx.base_uri}/api/markets'
+    try:
+        response = requests.get(request_url)
+    except requests.exceptions.RequestException as e:
+        raise RemoteError(
+            f'FTX get request at {request_url} connection error: {str(e)}.',
+        ) from e
+
+    if response.status_code != HTTPStatus.OK:
+        raise RemoteError(
+            f'FTX query responded with error status code: {response.status_code} '
+            f'and text: {response.text}',
+        )
+    try:
+        response_dict = jsonloads_dict(response.text)
+    except JSONDecodeError as e:
+        raise RemoteError(f'FTX returned invalid JSON response: {response.text}') from e
+
+    # Extract the unique symbols from the exchange pairs
+    unknown_assets = set()
+    unsupported_assets = set()
+    for entry in response_dict['result']:
+        if entry['type'] == 'future':
+            continue
+
+        base_currency = entry.get('baseCurrency', '')
+        quote_currency = entry.get('quoteCurrency', '')
+        try:
+            if base_currency not in unknown_assets:
+                asset_from_ftx(base_currency)
+        except UnsupportedAsset:
+            assert base_currency in unsupported_assets
+        except UnknownAsset as e:
+            test_warnings.warn(UserWarning(
+                f'Found unknown asset {e.asset_name} in FTX. '
+                f'Support for it has to be added',
+            ))
+            unknown_assets.add(base_currency)
```

Comment:
Make this into a helper function. Then run it 2 times. Once for base, and once for quote currency.